internal/util/projutil/exec.go

```diff
@@ -130,14 +126,47 @@ func wdInGoPath() (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	hd, err := homedir.Dir()
+	hd, err := getHomeDir()
 	if err != nil {
 		return false, err
 	}
-	if hd, err = homedir.Expand(hd); err != nil {
-		return false, err
-	}
 	goPath, ok := os.LookupEnv(GoPathEnv)
 	defaultGoPath := filepath.Join(hd, "go")
 	return (!ok && strings.HasPrefix(wd, defaultGoPath)) || (goPath != "" && strings.HasPrefix(wd, goPath)), nil
 }
+
+// From https://github.com/golang/go/wiki/Modules:
+//	You can activate module support in one of two ways:
+//	- Invoke the go command in a directory outside of the $GOPATH/src tree,
+//		with a valid go.mod file in the current directory or any parent of it and
+//		the environment variable GO111MODULE unset (or explicitly set to auto).
+//	- Invoke the go command with GO111MODULE=on environment variable set.
+//
+// GoModOn returns true if go modules are on in one of the above two ways.
+func GoModOn() (bool, error) {
+	inSrc, err := wdInGoPathSrc()
+	if err != nil {
+		return false, err
+	}
+	v, ok := os.LookupEnv(GoModEnv)
+	if v == "off" {
+		return false, nil
+	}
+	return (v == "on" && inSrc) || ((!ok || v == "" || v == "auto") && !inSrc), nil
```

Comment:
I think you have a logic bug where for the case where `v == "on" && !inSrc`. In that case, it looks like we return false when I think we should return true.

I'd also suggest re-organizing to avoid checking if we're in GOPATH if we don't have to.

```go
func GoModOn() (bool, error) {
	v, ok := os.LookupEnv(GoModEnv)
	if v == "off" {
		return false, nil
	}
	if v == "on" {
		return true, nil
	}
	inSrc, err := wdInGoPathSrc()
	if err != nil {
		return false, err
	}
	return !inSrc && (!ok || v == "" || v == "auto"), nil
}
```